.github/workflows/generate-ts-declaration.yml

```
@@ -0,0 +1,48 @@
+name: generate-typescript-declaration
+run-name: generate-typescript-declaration
+on:
+  push:
+    paths:
+      - '**.schema.json'
+
+jobs:
+  generate-typescript-declaration:
+    runs-on: ubuntu-latest
+    name: generate-typescript-declaration
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v3
+
+      - name: Generate application schemas to TypeScript declarations
+        uses: ./.github/actions/generate-typescript-declaration
+        with:
+          schema-directory: ./schemas/applications/schema
+          schema: application.schema.json
+          ts-declaration: application.d.ts
+
+      - name: Generate component schemas to TypeScript declarations
+        uses: ./.github/actions/generate-typescript-declaration
+        with:
+          schema-directory: ./schemas/component-descriptions/schema
+          schema: component.schema.json
+          ts-declaration: dynamic-component-description.d.ts
+
+      - name: Checkout frontend repository
+        uses: actions/checkout@v3
+        with:
+          repository: aica-technology/frontend
+          ref: develop
+          path: frontend
+          token: ${{ secrets.FRONTEND_REPOSITORY_WRITE }}
```

According to https://github.com/peter-evans/create-pull-request/issues/48#issuecomment-584209254, if we use SSH deploy keys, then the push can be used to trigger GitHub actions.

```suggestion
          ssh-key: ${{ secrets.FRONTEND_REPOSITORY_WRITE_SSH }}
```